experimental/Schemes/elliptic_surface.jl

```diff
@@ -52,30 +52,42 @@ base_ring(X::EllipticSurface) = coefficient_ring(base_ring(base_field(generic_fi
 @doc raw"""
     set_mordell_weil_basis!(X::EllipticSurface, mwl_basis::Vector{EllipticCurvePoint})
 
-Set generators for the Mordell-Weil lattice of ``X`` or at least of a sublattice.
+Set a basis for the Mordell-Weil lattice of ``X`` or at least of a sublattice.
 
 This invalidates previous computations depending on the generators of the
-mordell weil lattice such as the `algebraic_lattice`. Use with care.
+Mordell Weil lattice such as the `algebraic_lattice`. Use with care.
+
+The points in `mwl_basis` must be linearly independent.
 """
 function set_mordell_weil_basis!(X::EllipticSurface, mwl_basis::Vector{<:EllipticCurvePoint})
   @req all(parent(P) == generic_fiber(X) for P in mwl_basis) "points must lie on the generic fiber"
   X.MWL = mwl_basis
   # clear old computations
   if has_attribute(X, :algebraic_lattice)
-    deleteat!(X.__attrs, :algebraic_lattice)
+    delete!(X.__attrs, :algebraic_lattice)
   end
   if has_attribute(X, :mordell_weil_lattice)
-    deleteat!(X.__attrs, :mordell_weil_lattice)
+    delete!(X.__attrs, :mordell_weil_lattice)
   end
 end
 
 @doc raw"""
-    elliptic_surface(generic_fiber::EllipticCurve, euler_characteristic::Int, mwl_basis::Vector{<:EllipticCurvePoint}=EllipticCurvePoint[]) -> EllipticSurface
+    elliptic_surface(generic_fiber::EllipticCurve,
+    euler_characteristic::Int, mwl_gens::Vector{<:EllipticCurvePoint}=EllipticCurvePoint[];
+    is_basis::Bool=true)
+    -> EllipticSurface
```

Comment:
On my screen the indentation looks wrong (arguments not indented more than the name of the function), please check again. But my display settings might be off...